elimu_yangu/profiles/census.py

```diff
@@ -31,79 +36,112 @@ def get_profile(geo, profile_name, request, year):
-                    .filter(Base.metadata.tables['secondary_school'].c.more_than_40.like("yes%"))\
-                    .order_by(asc(cast(rank_column, Integer)))\
-                    .all()
-    # Getting top for schools with less than 40 students
-    top_schools_40_less = session.query(Base.metadata.tables['secondary_school'])\
-                    .filter(Base.metadata.tables['secondary_school'].c.geo_level == geo.geo_level)\
-                    .filter(Base.metadata.tables['secondary_school'].c.geo_code == geo.geo_code)\
-                    .filter(Base.metadata.tables['secondary_school'].c.year_of_result == year)\
-                    .filter(Base.metadata.tables['secondary_school'].c.more_than_40.like("no%"))\
-                    .order_by(asc(cast(rank_column, Integer)))\
-                    .all()
-
-    # Getting lowest schools with more than 40 students
-    lowest_schools_40_more = session.query(Base.metadata.tables['secondary_school'])\
-                    .filter(Base.metadata.tables['secondary_school'].c.geo_level == geo.geo_level)\
-                    .filter(Base.metadata.tables['secondary_school'].c.geo_code == geo.geo_code)\
-                    .filter(Base.metadata.tables['secondary_school'].c.year_of_result == year)\
-                    .filter(Base.metadata.tables['secondary_school'].c.more_than_40.like("yes%"))\
-                    .order_by(desc(cast(rank_column, Integer)))\
-                    .all()
-    # Getting lowest for schools with less than 40 students
-    lowest_schools_40_less = session.query(Base.metadata.tables['secondary_school'])\
-                    .filter(Base.metadata.tables['secondary_school'].c.geo_level == geo.geo_level)\
-                    .filter(Base.metadata.tables['secondary_school'].c.geo_code == geo.geo_code)\
-                    .filter(Base.metadata.tables['secondary_school'].c.year_of_result == year)\
-                    .filter(Base.metadata.tables['secondary_school'].c.more_than_40.like("no%"))\
-                    .order_by(desc(cast(rank_column, Integer)))\
-                    .all()
-    # median gpa
-    db_model_age = get_model_from_fields(['code', 'name', 'avg_gpa'], geo.geo_level)
-    objects = get_objects_by_geo(db_model_age, geo, session, ['avg_gpa'])
-    median = calculate_median(objects, 'avg_gpa')
-
-    # gpa in 1 point groups
-    def gpa_recode(f, x):
-        gpa = x
-        if gpa >= 4:
-            return '4+'
-        bucket = 1 * (gpa / 1)
-        return '%d-%d' % (bucket, bucket + 2)
-
-    gpa_dist_data, total_schools = get_stat_data(
-        'avg_gpa', geo, session,
-        table_fields=['code', 'name', 'avg_gpa'],
-        recode=gpa_recode, exclude=['unspecified'], only={'year_of_result': [year]})
-
-    total_private = 0.0
-    for data in schools_dist.get('Non-Government', {}).itervalues():
-        if 'numerators' in data:
-            total_private += data['numerators']['this']
+    with dataset_context(year='2017'):
```

Comment:
Just a question, is it possible to use `latest` here than a specific year?

Reply:
Something to consider later instead?

Reply:
Yes, sure, if it requires any work on our part, we can and should do it later. I thought `wazimap/HURUmap` converts `latest` to whichever year is the latest in the list of `available_release_years` for us automatically.